src/sentry/incidents/subscription_processor.py

```diff
@@ -633,19 +634,65 @@ def trigger_resolve_threshold(
     def handle_trigger_actions(
         self, incident_triggers: List[IncidentTrigger], metric_value: float
     ) -> None:
-        actions = deduplicate_trigger_actions(triggers=deepcopy(incident_triggers))
+        actions = deduplicate_trigger_actions(triggers=deepcopy(self.triggers))
         # Grab the first trigger to get incident id (they are all the same)
         # All triggers should either be firing or resolving, so doesn't matter which we grab.
         incident_trigger = incident_triggers[0]
-        method = "fire" if incident_triggers[0].status == TriggerStatus.ACTIVE.value else "resolve"
+        method = "fire" if incident_trigger.status == TriggerStatus.ACTIVE.value else "resolve"
+        try:
+            incident = Incident.objects.get(id=incident_trigger.incident_id)
+        except Incident.DoesNotExist:
+            metrics.incr("incidents.alert_rules.action.skipping_missing_incident")
+            return
+
+        incident_activities = IncidentActivity.objects.filter(incident=incident)
+        past_statuses = {int(i.value) for i in incident_activities if i.value is not None}
```

Comment:
There are  some incidents with 1000s of activity rows. It's not normal but can happen, usually on alerts that flip flop a lot but never resolve. Can we limit this to some reasonable number?

Since I think all we care about is the statuses here, we could just use `values_list` to fetch the value, and the use `.distinct` to get the limited number of statuses